### ISSUE TITLE
itk python `GetImageFromArray` should be invariant to the numpy memory layout changes?

### ISSUE THREAD

**Issue by wyli** [InsightSoftwareConsortium/ITK#3044]
### Description
`GetImageFromArray` checks the `C/F_CONTIGUOUS` flags for the channel dimension, but the spatial dimensions are always assumed C-indexing, this is somewhat inconsistent.

```py
import numpy as np
import itk

np_img = np.array([[1., 2.], [3., 4.], [5., 6.]])

np_c = np_img.copy(order="C")
np_f = np_img.copy(order="F")
print(np_c.shape, np_f.shape)

itk_c = itk.GetImageFromArray(np_img.copy(order="C"))
itk_f = itk.GetImageFromArray(np_img.copy(order="F"))
print(itk_c.shape, itk_f.shape)
```

### Expected behavior

`itk_c.shape == itk_f.shape`, because for the underlying numpy array, we have
```py
>>> np_c - np_f
array([[0., 0.],
       [0., 0.],
       [0., 0.]])
```

### Actual behavior

the outputs of the script are
```none
(3, 2) (3, 2)
(3, 2) (2, 3)
```

### Versions
the example is evaluated with itk-5.3rc2.post1

**Comment by thewtex** [comment]
@wyli thanks for the report. Addressed in #3098 , but note that the change only applies with `itk.GetImageViewFromArray`. `itk.GetImageFromArray` creates a new array that is necessarily C-order.

**Comment by wyli** [comment]
thank you @thewtex!
